src/mutations/updateGroupsForAccounts.js

```diff
@@ -0,0 +1,105 @@
+import ReactionError from "@reactioncommerce/reaction-error";
+import SimpleSchema from "simpl-schema";
+
+const inputSchema = new SimpleSchema({
+  accountIds: {
+    type: Array
+  },
+  "accountIds.$": String,
+  groupIds: {
+    type: Array
+  },
+  "groupIds.$": String
+});
+
+/**
+ * @name accounts/updateGroupsForAccounts
+ * @memberof Mutations/Accounts
+ * @method
+ * @summary Bulk-update group assignments for specified accounts
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Input arguments
+ * @param {String} input.accountIds - The account IDs
+ * @param {String} input.groupIds - The group IDs
+ * @return {Promise<Object>} accounts with updated groups
+ */
+export default async function updateGroupsForAccounts(context, input) {
+  inputSchema.validate(input);
+
+  const { accountIds, groupIds } = input;
+  const {
+    appEvents,
+    collections: {
+      Accounts,
+      Groups
+    },
+    userId
+  } = context;
+
+  const groupsToAssignAccountsTo = await Groups.find({
+    _id: {
+      $in: groupIds
+    }
+  }).toArray();
+
+  if (!groupsToAssignAccountsTo) {
+    throw new ReactionError("not-found", "No groups matching the provided IDs were found");
+  }
+
+  if (groupsToAssignAccountsTo.length !== groupIds.length) {
+    throw new ReactionError("not-found", `Could not find ${groupIds.length - groupsToAssignAccountsTo.length} of ${groupIds.length} groups provided`);
+  }
+
+  const shopIds = groupsToAssignAccountsTo.reduce((allShopIds, group) => {
+    if (!allShopIds.includes(group.shopId)) {
```

Comment:
I actually thought about it, and I don't think anyone's going to update hundreds of groups for hundreds of users, where it would really make a difference in performance... What do you think?

I've pushed a new iteration of this which simply iterates on `groupsToAssignAccountsTo` and checks for permissions whether the specific `shopId` was already checked or not. It also checks for global permissions if a group doesn't have a `shopId`. Let me know if this is acceptable solution of if you really want to avoid checking already-checked `shopId`s.